src/Common/tests/Cryptography/AlgorithmImplementations/AES/AesCornerTests.cs

```diff
@@ -0,0 +1,328 @@
+                {
+                    Assert.True(encryptor.CanTransformMultipleBlocks);
+                    byte[] cipher = encryptor.TransformFinalBlock(plainText, 0, plainText.Length);
+                    Assert.Equal<byte>(expectedCipher, cipher);
+                }
+
+                using (ICryptoTransform decryptor = a.CreateDecryptor())
+                {
+                    Assert.True(decryptor.CanTransformMultipleBlocks);
+                    byte[] decrypted = decryptor.TransformFinalBlock(expectedCipher, 0, expectedCipher.Length);
+                    Assert.Equal<byte>(plainText, decrypted);
+                }
+            }
+        }
+
+        [Fact]
+        public static void FinalOnlyTransformPKCS7()
+        {
+            // Use no TransformBlock calls() - do the entire transform using only TransformFinalBlock().
+            byte[] key = "1ed2f625c187b993256a8b3ccf9dcbfa5b44b4795c731012f70e4e64732efd5d".HexToByteArray();
+            byte[] iv = "47d1e060ba3c8643f9f8b65feeda4b30".HexToByteArray();
+
+            byte[] plainText = "f238882f6530ae9191c294868feed0b0df4058b322377dec14690c3b6bbf6ad1dd5b7c063a28e2cca2a6dce8cc2e668ea6ce80cee4c1a1a955ff46c530f3801b".HexToByteArray();
+            // CBC, Padding.PKCS7
+            byte[] expectedCipher = "7c6e1bcd3c30d2fb2d92e3346048307dc6719a6b96a945b4d987af09469ec68f5ca535fab7f596fffa80f7cfaeb26eefaf8d4ca8be190393b2569249d673f042a6a223f1c1069aa1d3c19d6bc454c205".HexToByteArray();
+
+            using (Aes a = Aes.Create())
+            {
+                a.Key = key;
+                a.IV = iv;
+                a.Mode = CipherMode.CBC;
+                a.Padding = PaddingMode.PKCS7;
+
+                using (ICryptoTransform encryptor = a.CreateEncryptor())
+                {
+                    Assert.True(encryptor.CanTransformMultipleBlocks);
+                    byte[] cipher = encryptor.TransformFinalBlock(plainText, 0, plainText.Length);
+                    Assert.Equal<byte>(expectedCipher, cipher);
+                }
+
+                using (ICryptoTransform decryptor = a.CreateDecryptor())
+                {
+                    Assert.True(decryptor.CanTransformMultipleBlocks);
+                    byte[] decrypted = decryptor.TransformFinalBlock(expectedCipher, 0, expectedCipher.Length);
+                    Assert.Equal<byte>(plainText, decrypted);
+                }
+            }
+        }
+
+        [Fact]
+        public static void ZeroLengthTransformNoPad()
+        {
+            // Use no TransformBlock calls() - do the entire transform using only TransformFinalBlock().
+            byte[] key = "1ed2f625c187b993256a8b3ccf9dcbfa5b44b4795c731012f70e4e64732efd5d".HexToByteArray();
+            byte[] iv = "47d1e060ba3c8643f9f8b65feeda4b30".HexToByteArray();
+
+            byte[] plainText = "".HexToByteArray();
+            // CBC, Padding.None
+            byte[] expectedCipher = "".HexToByteArray();
```

Comment:
Symmetry with the other crytotests that use the <string>.ToHexByteArray() pattern.